flask_dance/contrib/google.py

```diff
@@ -49,6 +49,41 @@ def make_google_blueprint(
         backend: A storage backend class, or an instance of a storage
                 backend class, to use for this blueprint. Defaults to
                 :class:`~flask_dance.consumer.backend.session.SessionBackend`.
+        hosted_domain (str, optional): The domain of the G Suite user. Used to indicate that the account selection UI
+            should be optimized for accounts at this domain. Note that this only provides UI optimization, and requires
+            response validation (see warning).
+
+    .. _google_hosted_domain_warning:
+    .. warning::
+       The ``hosted_domain`` argument **only provides UI optimization**. Don't rely on this argument to control
+       who can access your application. You must verify that the ``hd`` claim of the response ID token matches the
+       ``hosted_domain`` argument passed to ``make_google_blueprint``. For example:
+
+       .. code-block:: python
+
+            from flask import session, abort
+            from flask_dance.consumer import oauth_authorized
+            from flask_dance.contrib.google import make_google_blueprint, google
+            import requests
+
+            google_bp = make_google_blueprint(
+                client_id="foo",
+                client_secret="bar",
+                scope=["profile", "email"],
+                hosted_domain="example.com"
+            )
+
+            @oauth_authorized.connect_via(google_bp)
+            def logged_in(blueprint, token):
+                resp_json = google.get("/oauth2/v2/userinfo").json()
+                if resp_json["hd"] != blueprint.authorization_url_params["hd"]:
+                    google.authenticated = False
+                    requests.post("https://accounts.google.com/o/oauth2/revoke",
+                                  params={"token": token["access_token"]},
+                                  headers={"content-type": "application/x-www-form-urlencoded"})
```

Comment:
Is this header actually needed? If so, we should leave it in, but if it's not actually necessary, we should take it out, so that the docs are as simple as possible.

Reply:
Nope. This header was not needed. Removed in https://github.com/singingwolfboy/flask-dance/pull/166/commits/676176a67edd9db77b09879647cac5db742d90c9.